hyahatiph-labs/onion-monero-blockchain-explorer
# [WIP] add ASaaS (atomic swaps as a service) poc

Create a wrapper around the [comit-network](https://github.com/comit-network/xmr-btc-swap/) swap binary. 

## Steps to test
1. build as normal
2. pull the latest swap binary into the build directory- `wget https://github.com/comit-network/xmr-btc-swap/releases/download/0.10.2/asb_0.10.2_,< *PLATFORM* >.tar`
3. unpack with `tar xvf asb*.tar`
4. start xmrblocks with stagenet flag and the newly added `--enable-swaps` flag, click 'Atomic Swaps' link
5. optional - custom rendezvous point with `--rendezvous-point` flag

*see this [link](https://blog.sethforprivacy.com/guides/run-an-atomic-swap-provider-advanced/) for steps on how to configure asb. update monero rpc for stagenet and bitcoin electrum rpc url for testnet


## TODOs

- [x] add `--enable-swaps` flag
- [x] make rendezvous-point configurable at cmdline
- [x] create entry point for selected seller, addresses , etc.
- [x] make commands mutate based on stagenet flag
- [x] create page for the swap status
- [ ] use client_id until swap starts, allow user to obtain swap_id
- [ ] use swap_id for history, resume, cancel etc. functionality
- [ ] deploy code
- [ ] open bug bounty request / review


![omb-comit](https://user-images.githubusercontent.com/13033037/147397847-1ecfbde0-163a-499b-a3e0-7c98ca673269.png)
